src/PostgREST/App.hs

```diff
@@ -72,13 +77,16 @@ postgrest conf dbStructure pool =
     time <- getPOSIXTime
     body <- strictRequestBody req
 
-    let handleReq = runWithClaims conf time (app dbStructure conf body) req
+    let schema = cs $ configSchema conf
+        apiRequest = userApiRequest schema req body
+        handleReq = runWithClaims conf time (app dbStructure conf apiRequest) req
+
     resp <- either pgErrResponse id <$> P.use pool
-      (HT.run handleReq HT.ReadCommitted HT.Write)
+      (HT.run handleReq HT.ReadCommitted (transactionMode $ iAction apiRequest))
     respond resp
 
-app :: DbStructure -> AppConfig -> RequestBody -> Request -> H.Transaction Response
-app dbStructure conf reqBody req =
+app :: DbStructure -> AppConfig -> ApiRequest -> Request -> H.Transaction Response
+app dbStructure conf apiRequest req =
```

Comment:
I did exactly that but then changed it back since runWithClaims expects an "app" with a signature that accepts a req parameter, so anyway i was passiing that parameters and adding a "_" in it's place, i.e. a lot more places would have needed changes (the cleanup i described above) but i tried to keep the changes as minimal as possible.
On the next refactoring this can be done